RFC.md

```diff
@@ -1,19 +1,189 @@
+	}
+});
+```
+
+**Amplify v6 (`aws-amplify@6`)**
+
+```Typescript
+import {Hub} from "aws-amplify"
+
+const channel = "auth"
+const signInFailureEvent = "signInFailure"
+const signInFailureData = new AuthError("sign-in failed")
+
+// By default, the dispatch API will provide a list of all Amplify supported channels.
+// Addionally, it shows all the events and data associated to a given channel.
+Hub.dispatch(channel,
+ {
+  event: signInFailureEvent ,
+  data: "string" // it errors as "string" is not AuthError
+} )
+
+// It listens for events associated to a channel and infers the right payload data on the switch statement.
+Hub.listen(channel, ({ payload }) => {
+	switch (payload.event) { // event - signIn | signInFailure | signUp | signUpFailure | ...etc
+		case "signInFailure":
+			const data = payload.data; // data - AuthError
+			break;
+	}
+});
+```
+
+## Typescript support for custom Hub channels
+
+We are introduccing strong type inference for custom hub channels and payload.
+
+**Amplify v5 (`aws-amplify@5`)**
+
+```Typescript
+import {Hub} from "aws-amplify"
+
+const channel = "custom"
+const customEvent = "custom_event"
+const customData = "custom_data"
+
+Hub.dispatch(channel,{
+  event: customEvent,
+  data: customData
+  })
+
+Hub.listen(channel, ({ payload }) => {
+	switch (payload.event) { // event - string
+		case customEvent:
+			const data = payload.data; // data - any
+			break;
+	}
+});
+```
+
+**Amplify v6 (`aws-amplify@6`)**
+
+```Typescript
+import {Hub} from "aws-amplify"
+
+// name of the channel
+type channel = "custom"
+
+// Each key in the map represents a payload event and the key value is the data type for that event.
+// Note: If an event is assigned the null type, the payload object will not contain a data key.
+type customEventDataMap = {
+  A: number;
+  B: string;
+  C: null;
+  D: Object
+}
+
+Hub.dispatch<channel, customEventDataMap >("custom", {
+  event: "A" , // Union type of all possible events A | B | C | D
+  data: "string" // It errors as "string" is not a number
+})
+
+Hub.listen<channel, customEventDataMap>("custom", ({payload})=>{
+
+  switch(payload.event) // Union type of all possible events A | B | C | D
+  {
+    case "A":
+      payload.data; // data - number
+      break
+    case "B":
+      payload.data // data - string
+      break
+    case "C":
+      payload.data // it errors, there is no a data key on the payload object
+      break
+    case "D":
+      payload.data // data - Object
+      break
+  }
+})
+```
+
+## Typescript support Configuration
+
+In order to help developers to configure Amplify categories, we are introducing type support for the `Amplify.configure` API.
+This change provides IntelliSense on the configuration for each category. Example bellow shows an `Auth` configuration.
+
+**Amplify v5 (`aws-amplify@5`)**
+
+```Typescript
+
+import {Amplify} from "aws-amplify"
+
+const authConfig = {
+  userPoolId: "us-east-1_aaaaaaa",
+  userPoolClientId: "bbbbbbbbbbbbbbbbbb",
```

Comment:
```suggestion
  userPoolId: "us-east-1_0yqxxHm5q",
  userPoolClientId: "3keodiqtm52nhh2ls0vQfs5v1q",
```

Just used random values to make the example look more realistic, maybe not required?